### PR TITLE
fix: read_json for edges cases between nrows, lines=True TCTC-1973

### PR DESCRIPTION
**Pull request by Sanix-Darker**
## WHAT

- fix retro-compatibility with read_json
- handle edge cases with nrows and 'lines=True/False'
- added tests for the json reader

Screenshot from 2022-03-03 11-06-41

### CODE REVIEW

File: peakina/readers/json.py [hunk 1]
```diff
@@ -29,17 +29,23 @@ def read_json(
...
-    # for the preview_nrows and the preview_offset, we're going to convert in to list here
-    if preview_nrows is not None:
-        # In case we don't have the native nrows given in kwargs, we're going
-        # to use the provided preview_nrows
-        if (nrows := kwargs.get("nrows")) is None:
-            nrows = preview_nrows
-
-        kwargs["nrows"] = nrows
+    if filter is None:
+        filter = "."
```
Comment (raphaelvignes) [inline peakina/readers/json.py:33]: :+1:

File: peakina/readers/json.py [hunk 2]
```diff
@@ -29,17 +29,23 @@ def read_json(
...
+
+    with open(path_or_buf, encoding=encoding) as f:
+        path_or_buf = transform_with_jq(f.read(), filter)
+
+    # In case we don't have the native nrows given in kwargs, we're going
+    # to use the provided preview_nrows
+    if (nrows := kwargs.get("nrows", preview_nrows)) is not None:
+        if kwargs.get("lines") and kwargs.get("lines") is True:
```
Comment (raphaelvignes) [inline peakina/readers/json.py:41]:
```suggestion
        if (lines := kwargs.get("lines")) is True:
```

**Review by raphaelvignes — APPROVED** [review]
Ptite suggestion. Merci pour cette PR :+1:

File: peakina/readers/json.py (hunk 2 above)
Comment (Sanix-Darker) [reply peakina/readers/json.py:41]: Merci, pour cette suggestion, mais je me dis que vu que cette variable n'ai pas utilisee plus bas... j'aurais le linter qui va faire des caprices...

Comment (Sanix-Darker) [reply peakina/readers/json.py:41]: ah... as expected..
Screenshot from 2022-03-03 13-20-34
.
